priam/src/main/java/com/netflix/priam/resources/RestoreServlet.java

```diff
@@ -160,25 +164,12 @@ private void restore(String token, String region, Date startTime, Date endTime,
             restoreObj.restore(startTime, endTime);
         } finally {
             config.setDC(origRegion);
-            priamServer.getId().getInstance().setToken(origToken);
+            priamServer.getId().setBackupIdentifier(origBackupIdentifier);
         }
         tuner.updateAutoBootstrap(config.getYamlLocation(), false);
         cassProcess.start(true);
     }
 
-    /**
-     * Find closest token in the specified region
-     */
-    private String closestToken(String token, String region) {
```

Comment:
I moved this into RestoreTokenSelector to keep the logic for determining the closest token in one place